src/explorer.api/Models/ExploreResult.cs

```diff
@@ -8,25 +8,56 @@ namespace Explorer.Api.Models
 
     internal class ExploreResult
     {
-        public ExploreResult(Guid explorationId, ExplorationStatus status)
+        public ExploreResult(Guid explorationId, ExplorationStatus status, string dataSource, string table)
         {
             Id = explorationId;
             Status = status;
-            Metrics = Array.Empty<Metric>();
+            DataSource = dataSource;
+            Table = table;
+            Columns = Array.Empty<ColumnMetricsCollection>();
+            SampleData = Array.Empty<IEnumerable<object?>>();
         }
 
-        public ExploreResult(Guid explorationId, ExplorationStatus status, IEnumerable<Metric> metrics)
+        public ExploreResult(
+            Guid explorationId,
+            ExplorationStatus status,
+            string dataSource,
+            string table,
+            IEnumerable<ColumnMetricsCollection> columnMetrics,
+            IEnumerable<IEnumerable<object?>> sampleData)
         {
             Id = explorationId;
             Status = status;
-            Metrics = metrics;
+            DataSource = dataSource;
+            Table = table;
+            Columns = columnMetrics;
+            SampleData = sampleData;
         }
 
+        public Guid Id { get; }
+
         public ExplorationStatus Status { get; }
 
-        public IEnumerable<Metric> Metrics { get; }
+        public string DataSource { get; }
 
-        public Guid Id { get; }
+        public string Table { get; }
+
+        public IEnumerable<ColumnMetricsCollection> Columns { get; }
+
+        public IEnumerable<IEnumerable<object?>> SampleData { get; }
+
+        public class ColumnMetricsCollection
+        {
+            public ColumnMetricsCollection(string column, IEnumerable<Metric> metrics)
+            {
+                Column = column;
```

Comment:
When including this I also thought about #169 and from my point of view, this is equivalent the suggestion you made there.

Reply:
The point of #169 was also have extra information to help with debugging. For example, if we are looking at the list of metrics, we can see immediately what dataset / table / column it was generated from. Ideally we should also have something identifying the current release version (like a commit hash) although this is little trickier to add... 

To be honest it probably makes sense to have both (the extra metric debugging and the `Column` member in `ColumnMetricsCollection` to be included in the sample data.

Reply:
I also think that it is still useful to have the column as I added it in the current implementation, since I suppose there might be some situation where this will simplify the parsing/searching used to process the results in a program.
I will add the additional metric in the separate issue #169., and I suppose that we can add the version information as a property in the `ExploreResult` class.